src/main/java/org/elasql/perf/tpart/metric/TransactionMetricRow.java

```diff
@@ -0,0 +1,111 @@
+package org.elasql.perf.tpart.metric;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.elasql.util.CsvRow;
+import org.vanilladb.core.server.task.Task;
+
+public abstract class TransactionMetricRow extends Task {
```

Comment:
I see what you were trying to do here, but you might have noticed that these three types of rows have exactly same code. It is no need to create three different classes for these metrics, because there is no difference between their data. In addition, it is not a good practice to create a super class just for splitting the code. We usually create a package for them.

My suggestion would be to use only one row class (which may be called `LongValueRow`), and reuse that class for all metrics. In this case, you will not need to split the row class out from the recorder.